Userland/Libraries/LibWeb/DOM/Element.cpp

```diff
@@ -878,12 +878,14 @@ JS::NonnullGCPtr<Geometry::DOMRectList> Element::get_client_rects() const
     if (auto const* paintable_box = this->paintable_box())
         transform = Gfx::extract_2d_affine_transform(paintable_box->transform());
     CSSPixelPoint scroll_offset;
-    for (auto const* containing_block = paintable()->containing_block(); containing_block; containing_block = containing_block->containing_block()) {
-        transform = Gfx::extract_2d_affine_transform(containing_block->transform()).multiply(transform);
-        scroll_offset.translate_by(containing_block->scroll_offset());
+    auto const* paintable = this->paintable();
+    if (paintable) {
+        for (auto const* containing_block = paintable->containing_block(); containing_block; containing_block = containing_block->containing_block()) {
+            transform = Gfx::extract_2d_affine_transform(containing_block->transform()).multiply(transform);
+            scroll_offset.translate_by(containing_block->scroll_offset());
+        }
     }
```

Comment:
element that has layout node but does have paintable, means a mistake happened somewhere earlier in tree building, layout or somewhere else.
let's not cover it by adding null check.

Reply:
Is this fully true? I added a FIXME along similar lines for colgroup, but that FIXME was removed in: https://github.com/SerenityOS/serenity/commit/3591a82e8d5c62497f0202118efb9cecffd89eb5

(I may be misunderstanding something here though)

Reply:
well, then there was a mistake in 3591a82e8d5c62497f0202118efb9cecffd89eb5
any element with layout node is expected to have a paintable. @awesomekling please correct me if I am wrong.

Reply:
Uhh that may have been a mistake. It didn't break any tests, do we have a way to illustrate some problem?

Reply:
in this specific case `<lineargradient id="test" display="none" />`
we are not supposed to create layout nodes for elements with `display: none` so instead of adding a missing paintable check, we need to find why we created a layout node for lineargradient.

Reply:
After testing every element (I think), I've found I can get the same crash with the following elements:
Elements within an `<svg>` (only after setting the `display` attribute to `none`):
* `<lineargradient>`
* `<radialgradient>`
* `<marker>`
* `<stop>`
* Any unknown element name e.g: `<foo>`

HTML elements (without changing any `display` properties):
* `<br>`
* `<slot>`

I also have a Domato test where the error is triggered by a nested MathML element, but I haven't been able to reduce that to a point that it's clear what the root cause is.

For now, I've removed this commit from the PR. I'll submit a separate PR with the appropriate fix(es) for this once I've investigated further.